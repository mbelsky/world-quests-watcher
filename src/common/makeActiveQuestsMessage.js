export function makeActiveQuestsMessage(quests) {
  if (quests.length === 0) {
    return undefined;
  }

  const strList = quests.map((quest) => {
    const { href, name } = quest;
    return `â€” "${name}" https://www.wowhead.com${href}\n`;
  });

  return `
Today active quests:

${strList.join("")}
`.trim();
}
