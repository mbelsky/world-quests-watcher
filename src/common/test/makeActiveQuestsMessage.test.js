import { makeActiveQuestsMessage } from "../makeActiveQuestsMessage.js";

const quests = [
  {
    endsAt: "10 hours",
    href: "/quest=40337",
    id: "40337",
    name: "Flummoxed",
    region: "na",
  },
  {
    endsAt: "11 hours",
    href: "/quest=40850",
    id: "40850",
    name: "Prisoners of Greystone",
    region: "na",
  },
];

test("makeActiveQuestsMessage", () => {
  expect(makeActiveQuestsMessage(quests)).toMatchSnapshot();
});
