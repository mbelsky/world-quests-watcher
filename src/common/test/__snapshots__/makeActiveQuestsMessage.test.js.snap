// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeActiveQuestsMessage 1`] = `
"Today active quests:

— \\"Flummoxed\\" https://www.wowhead.com/quest=40337
— \\"Prisoners of Greystone\\" https://www.wowhead.com/quest=40850"
`;
